qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1996 @@
+
+    Example Usage:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse3Q
+
+        backend = FakeOpenPulse3Q()
+
+        with pulse.build(backend) as pulse_prog:
+            # Delay for 100 cycles on qubits 0, 1 and 2.
+            regs = pulse.delay_qubits(100, 0, 1, 2)
+
+    .. note:: Requires the active builder context to have a backend set.
+
+    Args:
+        duration: Duration to delay for.
+        qubits: Physical qubits to delay on. Delays will be inserted based on
+            the channels returned by :func:`pulse.qubit_channels`.
+    """
+    qubit_chans = set(itertools.chain.from_iterable(qubit_channels(qubit) for
+                                                    qubit in qubits))
+    with align_left(), group():
+        for chan in qubit_chans:
+            delay(duration, chan)
+
+
+# Gate instructions ############################################################
+def call_gate(gate: circuit.Gate, qubits: Tuple[int, ...], lazy: bool = True):
+    """Call a gate and lazily schedule it to its corresponding
+    pulse instruction.
+
+    .. note::
+        Calling gates directly within the pulse builder namespace will be
+        deprecated in the future in favour of tight integration with a circuit
+        builder interface which is under development.
+
+    .. jupyter-kernel:: python3
+        :id: call_gate
+
+    Example Usage:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.circuit.library import standard_gates as gates
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        with pulse.build(backend) as pulse_prog:
+            pulse.call_gate(gates.CXGate(), (0, 1))
+
+    We can see the role of the transpiler in scheduling gates by optimizing
+    away two consecutive CNOT gates:
+
+    .. jupyter-execute::
+
+        with pulse.build(backend) as pulse_prog:
+            with pulse.transpiler_settings(optimization_level=3):
+                pulse.call_gate(gates.CXGate(), (0, 1))
+                pulse.call_gate(gates.CXGate(), (0, 1))
+
+        assert pulse_prog == pulse.Schedule()
+
+    .. note:: If multiple gates are called in a row they may be optimized by
+        the transpiler, depending on the
+        :func:`pulse.active_transpiler_settings``.
+
+    .. note:: Requires the active builder context to have a backend set.
+
+    Args:
+        gate: Circuit gate instance to call.
+        qubits: Qubits to call gate on.
+        lazy: If ``false`` the gate will be compiled immediately, otherwise
+            it will be added onto a lazily evaluated quantum circuit to be
+            compiled when the builder is forced to by a circuit assumption
+            being broken, such as the inclusion of a pulse instruction or
+            new alignment context.
+    """
+    _active_builder().call_gate(gate, qubits, lazy=lazy)
+
+
+def cx(control: int, target: int):
```

Comment:
I think these convenience methods (`cx`, `u1`, `u2`, `u3`) should be removed. You've already stated that they're about to be deprecated. These certainly look nice, and it does get me excited for when we have one, coherent builder. Adding these now will just cause trouble later.

Reply:
For now, they are a requested feature by the experimentalists. We will have to deprecate at a later date.